Add configuration for not installing header into a subdirectory and f…

sdroege opened the pull request:
…or using a pre-generated header

Fixes https://github.com/lu-zero/cargo-c/issues/101

sdroege commented on src/build.rs:
```diff
@@ -251,21 +288,34 @@ fn load_manifest_overrides(root_path: &PathBuf) -> anyhow::Result<Overrides> {
+                .get("name")
+                .or_else(|| capi.get("header_name"))
                 .map(|v| v.clone().try_into())
                 .unwrap_or(Ok(None))?,
```
It's not added but whatever extension is there already is replaced with ".h" like before, and if you don't provide an extension then it's added :) That behaviour didn't change compared to before, that's from the 3 `set_extension("h")` calls.

lu-zero replied:
that part is fine, but the code is not picking the name as you see with nnnoiseless (that's the only crate using the feature)

sdroege replied:
That was the case before but with the latest version here it seems to work fine for me. Problem was that it was going into the default case if the `capi.header` configuration did not exist at all, instead of using the fallback "header_name" from the `capi` configuration.

sdroege commented:
> https://github.com/jneem/nnnoiseless/ does not produce the right header with this patch looks like the compatibility code is missing?

Fixed, and also cleaned up things a bit and added documentation in the `README.md` about this.

lu-zero commented:
Great !

lu-zero reviewed (approved):
Looks good to me, I'll land it tomorrow morning.

sdroege commented:
> Looks good to me, I'll land it tomorrow morning.

Thanks, I'll take care of the other two PRs afterwards.
